design_drafts/transmission_loading_and_parametrization.md

```diff
@@ -0,0 +1,194 @@
+
+Common policies are to match `bar_joint` to `bar_actuator` or `joint1` to `joint1_motor`.
+
+### Joint and actuator tags
+
+There are some minor issues that cause confusion from time to time:
+* Why is the `hardwareInterface` tag paired with the joint? Could be defined for both if we want to allow different interfaces or if we want to ensure that they are not different for compatibility...
+* Why is the `mechanicalReduction` tag paired with the actuator? The reduction is applied to one of them when the information flows from actuator to controller but the inverse is applied when it flows from controller to actuator.
+
+
+
+## Proposals for ROS2 XML format
+
+The plugin-based system should be kept for transmissions. The tag `type` was replaced with `plugin` in the xml format for clarity and consistency with the rest of `ros2_control`.
+
+The parametrization should become simpler as it doesn't need to concern itself anymore with the joint command interface types as they are defined elsewhere.
+
+It is however still unclear if there is any use to define actuator names outside the transmission scope.
+
+Where to place `mechanical_reduction`?
+
+The mandatory parameters within the `<transmission>` tag should `<plugin>` and at least one `<joint>` and `<actuator>` tag.
+
+The parametrization of `<joint>` and `<actuator>` should contain `name` and `role` at all times but additional parameters should be allowed to be freely defined as different transmission may require/support different ones, let them report if they are unhappy with the configuration they were given.
+
+### Proposal 1
+
+Merging the current ros2_control xml format with the ROS1 transmissions:
+
+```xml
+  <ros2_control name="RRBotSystemPositionOnly" type="system">
+    <hardware>
+      <plugin>ros2_control_demo_hardware/RRBotSystemPositionOnlyHardware</plugin>
+      <param name="example_param_hw_start_duration_sec">2.0</param>
+      <param name="example_param_hw_stop_duration_sec">3.0</param>
+      <param name="example_param_hw_slowdown">2.0</param>
+    </hardware>
+    <joint name="joint1">
+      <command_interface name="position">
+        <param name="min">-1</param>
+        <param name="max">1</param>
+      </command_interface>
+      <state_interface name="position"/>
+    </joint>
+    <joint name="joint2">
+      <command_interface name="position">
+        <param name="min">-1</param>
+        <param name="max">1</param>
+      </command_interface>
+      <state_interface name="position"/>
+    </joint>
+
+    <transmission name="differential_transmission_1">
+      <plugin>transmission_interface/DifferentialTransmission</plugin>
+      <actuator name="joint1_motor">
+        <role>actuator1</role>
+      </actuator>
+      <actuator name="joint2_motor">
+        <role>actuator2</role>
+      </actuator>
+      <joint name="joint1">
+        <role>joint1</role>
+        <offset>0.5</offset>
+        <mechanical_reduction>10</mechanical_reduction>
```

Comment:
For consistency, should `role` and `mechanical_reduction` be `param`s instead of tags? What's the criteria to chose one over the other?

Reply:
I can move mechanical reduction there too, the debate about it's position still stands. Please check proposal 2, that's where I will make this change